src/core/geometry/qgsgeos.cpp

```diff
@@ -896,32 +896,24 @@ QgsGeometryEngine::EngineOperationResult QgsGeos::splitPolygonGeometry( GEOSGeom
     return InvalidBaseGeometry;
   }
 
+  // we will need prepared geometry for intersection tests
+  const_cast<QgsGeos *>( this )->prepareGeometry();
```

Comment:
might be a bit of a performance win if you move this up to about line 882, so that the prepared geometry could be used for the splitline/polygon intersects test which happens on line 884.